File: speed/mul2.py
from mybench import *

a = reshape(arrayrange(1000),(1000,1)).astype(Float64)
b = reshape(arrayrange(1000),(1,1000)).astype(Float64)

print "a.typecode:",a.typecode(),",  a.shape:",a.shape
print "b.typecode:",b.typecode(),",  b.shape:",b.shape
print "calculating  c = a*b ..."

def bench_body(a=a,b=b): c=a*b

bench_time(bench_body)

c = a*b
print "c.shape:", c.shape,"\n"

File: speed/mul2.rb
require 'mybench'

a = NArray.float(1000,1).indgen!
b = NArray.float(1,1000).indgen!
c = 0

print "a = "; p a;
print "b = "; p b;
print "calculating c = a*b ...\n"

bench_time{ c = a*b }

print "c = "; p c;
